src/main/java/jalau/at18/katas/game2048/team1/Main.java

```diff
@@ -0,0 +1,259 @@
+            printMatrix(newMatrix);
+        }
+        if (directionValue.equals("d")) {
+            System.out.println("Move rigth");
+            newMatrix = moveRigth(matrix);
+            newMatrix = addTile(newMatrix);
+            printMatrix(newMatrix);
+        }
+        return newMatrix;
+    }
+
+    public int[][] moveTop(int[][] matrix) {
+        for (int moves = 0; moves < MAX_ROW_MOVES; moves++) {
+            for (int row = 1; row < matrix.length; row++) {
+                for (int column = 0; column < matrix.length; column++) {
+                    if (matrix[row][column] != 0 && matrix[row - 1][column] == 0) {
+                        matrix[row - 1][column] = matrix[row][column];
+                        matrix[row][column] = 0;
+                    } else if (matrix[row][column] != 0 && matrix[row - 1][column] == matrix[row][column]) {
+                        matrix[row - 1][column] = 2 * (matrix[row][column]);
+                        matrix[row][column] = 0;
+                    }
+                }
+            }
+        }
+        return matrix;
+    }
+
+    public int[][] moveDown(int[][] matrix) {
+        for (int moves = 0; moves < MAX_ROW_MOVES; moves++) {
+            for (int row = 2; row >= 0; row--) {
+                for (int column = 0; column < matrix.length; column++) {
+                    if (matrix[row][column] != 0 && matrix[row + 1][column] == 0) {
+                        matrix[row + 1][column] = matrix[row][column];
+                        matrix[row][column] = 0;
+                    } else if (matrix[row][column] != 0 && matrix[row + 1][column] == matrix[row][column]) {
+                        matrix[row + 1][column] = 2 * (matrix[row][column]);
+                        matrix[row][column] = 0;
+                    }
+                }
+            }
+        }
+        return matrix;
+    }
+
+    public int[][] moveLeft(int[][] matrix) {
+        for (int moves = 0; moves < MAX_ROW_MOVES; moves++) {
+            for (int column = 1; column < matrix.length; column++) {
+                for (int row = 0; row < matrix.length; row++) {
+                    if (matrix[row][column] != 0 && matrix[row][column - 1] == 0) {
+                        matrix[row][column - 1] = matrix[row][column];
+                        matrix[row][column] = 0;
+                    } else if (matrix[row][column] != 0 && matrix[row][column - 1] == matrix[row][column]) {
+                        matrix[row][column - 1] = 2 * (matrix[row][column]);
+                        matrix[row][column] = 0;
+                    }
+                }
+            }
+        }
+        return matrix;
+    }
+
+    public int[][] moveRigth(int[][] matrix) {
+        for (int moves = 0; moves < MAX_ROW_MOVES; moves++) {
+            for (int column = 2; column >= 0; column--) {
+                for (int row = 0; row < matrix.length; row++) {
+                    if (matrix[row][column] != 0 && matrix[row][column + 1] == 0) {
```

Comment:
Logic is good, but it's missing one of the requirements of the game:
```
- A tile cannot be merged twice in one single movement
```

Here an example when running the program:
![image](https://user-images.githubusercontent.com/4606607/184510928-6652f49d-bb34-4c24-93b4-511e4b8eeb4f.png)

For row at the bottom, result should be
```
0   0   4   4
```
When tiles are moved, 2 and 2 are merged once and cannot be merged twice, but they are, because at the end, they become 8. They can become 8, but that should happen after moving the grid to the right again

Reply:
We already correct this